core/frontend/render-modes/render-mode-base.php

```diff
@@ -0,0 +1,106 @@
+<?php
+namespace Elementor\Core\Frontend\RenderModes;
+
+use Elementor\Plugin;
+use Elementor\Core\Base\Document;
+use Elementor\Core\Frontend\Render_Mode_Manager;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+abstract class Render_Mode_Base {
+	const ENQUEUE_SCRIPTS_PRIORITY = 10;
+	const ENQUEUE_STYLES_PRIORITY = 10;
+
+	/**
+	 * @var int
+	 */
+	protected $post_id;
+
+	/**
+	 * @var Document
+	 */
+	protected $document;
+
+	/**
+	 * Render_Mode_Base constructor.
+	 *
+	 * @param $post_id
+	 */
+	public function __construct( $post_id ) {
+		$this->post_id = intval( $post_id );
+	}
+
+	/**
+	 * Returns the key name of the class.
+	 *
+	 * @return string
+	 */
+	abstract public static function get_name();
+
+	/**
+	 * @param $post_id
+	 *
+	 * @return string
+	 */
+	public static function get_url( $post_id ) {
+		return Render_Mode_Manager::get_base_url( $post_id, static::get_name() );
+	}
+
+	/**
+	 * By default do not do nothing.
+	 */
+	public function prepare_render() {
+		add_action( 'wp_enqueue_scripts', function () {
+			$this->enqueue_scripts();
+		}, static::ENQUEUE_SCRIPTS_PRIORITY );
+
+		add_action( 'wp_enqueue_scripts', function () {
+			$this->enqueue_styles();
+		}, static::ENQUEUE_STYLES_PRIORITY );
```

Comment:
You can use the short syntax: `add_action( 'wp_enqueue_scripts', [ $this, 'enqueue_scripts' ], static::ENQUEUE_SCRIPTS_PRIORITY );`

Reply:
It's for keeping the target function protected.